src/core/DataFile.cpp

```diff
@@ -794,6 +794,45 @@ void DataFile::upgrade_0_4_0_rc2()
 }
 
 
+void DataFile::upgrade_1_0_99()
+{
+	const jo_id_t EO_ID_MSB = 1 << 23;
+	
+	QList<jo_id_t> idList;
+	findIds(documentElement(), idList);
+	
+	QDomNodeList list = elementsByTagName("ladspacontrols");
+	for(int i = 0; !list.item(i).isNull(); ++i)
+	{
+		for(QDomNode node = list.item(i).firstChild(); !node.isNull();
+			node = node.nextSibling())
+		{
+			QDomElement el = node.toElement();
+			QDomNode data_child = el.namedItem("data");
+			if(!data_child.isElement())
+			{
+				if (el.attribute("scale_type") == "log")
+				{
+					QDomElement me = createElement("data");
+					me.setAttribute("value", el.attribute("data"));
+					me.setAttribute("scale_type", "log");
+					
+					jo_id_t id;
+					for(jo_id_t tid = rand(); idList.contains(id = tid % EO_ID_MSB
+															  | EO_ID_MSB); tid++)
```

Comment:
The ids from loaded projects do not have `EO_ID_MSB` set, that would conflict with new objects. This id allocation from `ProjectJournal` should not be here; just assign any free id, that is, `tid = last_assigned + 1` starting from 0.

Assume tabs are 8 spaces. This line is difficult to see in the web interface.